File: MyApplication2/app/src/main/java/com/example/myapplication/DatePickerFragment.java
package com.example.myapplication;

import android.app.DatePickerDialog;

import android.app.Dialog;

import android.content.Context;

import android.os.Bundle;


import android.widget.DatePicker;


import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

import java.util.Calendar;





public class DatePickerFragment extends DialogFragment implements DatePickerDialog.OnDateSetListener {



    public interface DatePickerListener {

        void onDateSet(DatePicker datePicker, int year, int month, int day);

    }



    DatePickerListener mListener;



    @Override

    public void onAttach(Context context) {

        super.onAttach(context);

        try {

            mListener = (DatePickerListener) context;

        } catch (Exception e) {

            throw new ClassCastException(getActivity().toString() + " Must Implements DatePickerListener");

        }

    }



    @NonNull

    @Override

    public Dialog onCreateDialog(Bundle savedInstanceState) {

        Calendar cal = Calendar.getInstance();

        int year = cal.get(Calendar.YEAR);

        int month = cal.get(Calendar.MONTH);

        int day = cal.get(Calendar.DAY_OF_MONTH);

        return new DatePickerDialog(getActivity(), this, year, month, day);

    }


    @Override

    public void onDateSet(DatePicker datePicker, int i, int i1, int i2) {

        mListener.onDateSet(datePicker, i, i1, i2);

    }

}

File: loginpage/app/src/main/java/com/example/task/DatePickerFragment.kt
package com.example.task

import android.app.DatePickerDialog
import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.widget.DatePicker
import androidx.fragment.app.DialogFragment
import java.util.*

class DatePickerFragment : DialogFragment(),
    DatePickerDialog.OnDateSetListener {
    interface DatePickerListener {
        fun onDateSet(datePicker: DatePicker?, year: Int, month: Int, day: Int)
    }

    var mListener: DatePickerListener? = null
    override fun onAttach(context: Context) {
        super.onAttach(context)
        mListener = try {
            context as DatePickerListener
        } catch (e: Exception) {
            throw ClassCastException(activity.toString() + " Must Implements DatePickerListener")
        }
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val cal = Calendar.getInstance()
        val year = cal[Calendar.YEAR]
        val month = cal[Calendar.MONTH]
        val day = cal[Calendar.DAY_OF_MONTH]
        return DatePickerDialog(activity!!, this, year, month, day)
    }

    override fun onDateSet(datePicker: DatePicker, i: Int, i1: Int, i2: Int) {
        mListener!!.onDateSet(datePicker, i, i1, i2)
    }
}